libnavigation-core/src/main/java/com/mapbox/navigation/core/SetRoutesCallback.kt

```diff
@@ -0,0 +1,27 @@
+package com.mapbox.navigation.core
+
+import com.mapbox.navigation.base.route.NavigationRoute
+
+/**
+ * Interface definition for a callback that gets notified whenever routes
+ * passed to [MapboxNavigation.setNavigationRoutes] are set or ignored for some reason.
+ */
+interface SetRoutesCallback {
+
+    /**
+     * Invoked whenever the routes passed to [MapboxNavigation.setNavigationRoutes]
+     * are successfully set.
+     *
+     * @param routes List of routes that were successfully set.
+     */
+    fun onRoutesSetResult(routes: List<NavigationRoute>)
```

Comment:
Let's add the wrapper, it will potentially help to extend result with new data. 
```suggestion
    fun onSuccess(result: RoutesCallbackSuccess)
```
where 
```kotlin
class RoutesCallbackSuccess internal constructor(
val routes: List<NavigationRoute>
) {
 // equal, hasCode, toString
}
```

Reply:
Makes sense, I'll change it when we agree about the naming.